File: Leetcode/reverseList.swift
    func reverseList(head: ListNode?) -> ListNode? {
        if head == nil {
            return nil
        }
        var result : ListNode? 
        var p = head
        while let current = p {
            p = p!.next
            current.next = result
            result = current
        }
        return result
    }

File: Leetcode/reverseList.java
public ListNode reverseList(ListNode head) {
        if (head == null) {
            return null;
        }
        ListNode result = null;
        ListNode p = head;
        while (p != null ) {
            ListNode node = p;
            p = p.next;
            node.next = null;
            if (result == null) {
                result = node;
            } else {
                node.next = result;
                result = node;
            }
        }
        return result;
    }
